lib/core/unittest/CStaticThreadPoolTest.cc

```diff
@@ -18,7 +18,9 @@
 
 #include <atomic>
 #include <chrono>
+#include <mutex>
 #include <thread>
+#include <unordered_set>
```

Comment:
We're still using `boost::unordered_set` everywhere else.  I think it's best we stick with one across the codebase.  Using `std` here will just encourage mixing the two.

Reply:
See [f047acd](https://github.com/elastic/ml-cpp/pull/2232/commits/f047acdece21675beec11357850a4b73d3defdd6).